### DISCUSSION TITLE
How to use simple-icons python wrapper

### DISCUSSION THREAD

**Discussion by nhtnhan** [simple-icons/simple-icons#5475] [General]
Hi, I'm trying to use the python wrapper for simple-icons from [pypi simple-icons 0.1.0
](https://pypi.org/project/simple-icons/). I already installed it but couldn't find any references/documentation on how to use it. In detail, I'm hoping to use it for searching icons existing in simple-icons repo.

I would really appreciate any help!

**Comment by ericcornelissen** [comment]
Hi @nhtnhan :wave: Unfortunately the Python wrapper is not officially provided by us, instead it's a third-party package developed by @vigonotion. I myself have never tried it, and nor has any of the other @simple-icons/maintainers I think, so I'm afraid we cannot be of any help to you.

[The pypi page](https://pypi.org/project/simple-icons/) links to [the simple-icons fork](https://github.com/vigonotion/simple-icons) of @vigonotion, unless they responds here you're probably better of opening an issue there :slightly_smiling_face:

**Reply by nhtnhan** [reply]
Hi,  I see 😅. Thank you for your quick response! I'll look into those 🙂

**Comment by vigonotion** [comment]
Hi, yes that's my project. That library is under development and not working at all yet. I want to use it to improve the hass-simpleicons project. Unfortunately, I cannot give you an ETA for the release of that library, I don't currently have time to develop that library.

If you are a programmer yourself, you can help me develop that library if you like, PRs welcome 🤗

**Reply by nhtnhan** [reply]
For sure, I will look into it and see what I can do!💪

**Reply by fbernhart** [reply]
Maybe I could help out as well. I'm quite familiar with Python and one of the Simple Icons maintainers. :)

**Comment by fbernhart** [comment]
@nhtnhan Have you already looked at https://pypi.org/project/simpleicons (source code is here: https://github.com/xCloudzx/simpleicons)? Seems like it's another recently developed Python wrapper for Simple Icons, which already (kind of) works and has some basic features. I just quickly tested it.

@xCloudzx has developed it. And maybe all of you (including @vigonotion) can work together on it? 😊

**Reply by sachinraja** [reply]
Looking into updating it right now. I'm attempting to stay true to the API. Random, unfixable pylance errors are getting annoying though 😅.

**Comment by sachinraja** [comment]
I've finished updating [my package (simpleicons)](https://pypi.org/project/simpleicons/). Also [repo here](https://github.com/xCloudzx/simpleicons). It utilizes the same tricks that the JS package does by generating (.py) files. I think you will find that it stays quite close to the API in this package. The documentation is also very similar (it's all in the README).

**Reply by service-paradis** [reply]
Good job @xCloudzx, feel free to add it in Simple Icons Readme if you want!

**Reply by sachinraja** [reply]
I'll definitely send in a PR, thank you!

**Reply by fbernhart** [reply]
Already looks promising so far! 👍🏼  But there are two things that might need improvement in my opinion. I'll open an issue in your repo. :)
